src/csharp/Microsoft.Spark/Sql/Types/ComplexTypes.cs

```diff
@@ -200,6 +226,27 @@ public StructType(IEnumerable<StructField> fields)
                 new JProperty("type", TypeName),
                 new JProperty("fields", Fields.Select(f => f.JsonValue).ToArray()));
 
+        /// <summary>
+        /// Does this type need to convert between C# object and internal SQL object.
+        /// </summary>
+        /// <returns>True if conversion needed, false otherwise</returns>
+        internal override bool NeedConversion() => true;
+
+        /// <summary>
+        /// Converts an internal SQL object into a native C# object.
```

Comment:
Don't you need a comment specific for `StructType`?

Reply:
Removed interfaces and reverted code.